controller_manager/src/controller_manager.cpp

```diff
@@ -2119,4 +2127,30 @@ controller_interface::return_type ControllerManager::check_preceeding_controller
   return controller_interface::return_type::OK;
 };
 
+void ControllerManager::controller_activity_diagnostic_callback(
+  diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  // lock controllers
+  std::lock_guard<std::recursive_mutex> guard(rt_controllers_wrapper_.controllers_lock_);
```

Comment:
Is this mutex lock going to affect realtime performance?  (Sorry I'm late to review this)